UKHomeOffice/nmsw-ui
# NMSW-514 Delete uploaded supporting document

# Ticket

NMSW-514

----

## AC

Given a user has uploaded a file
When they click delete
Then the file is removed

----

## To test

**To test deleting previously uploaded file**
- Run app
- Sign in
- Create a declaration
- Complete General declaration FAL 1 section
- On the task list page click supporting documents link
- Add a couple of valid files (csv, doc, docm, docx, rtf, txt, xls, xlsm, xlsx, xltm, xltx, xlw or xml AND a file size less than 1MB)
- Click upload
- Refresh the page
- Click delete
- > You should see a successful delete request in the network tab

**To test deleting file in 'limbo' state**
- Add another valid file
- Click upload
- Do NOT refresh the page
- Click delete
- > You should see a successful delete request in the network tab

----

## Notes

- Currently the file will not disappear from the list because the BE the file gets deleted from S3 but does not get deleted from the database
